// nuxi/dist/shared/nuxi.c4d14678.mjs
const _rDefault = (r) => r.default || r;
const commands = {
  dev: () => import('../chunks/dev.mjs').then(_rDefault),
  build: () => import('../chunks/build.mjs').then(_rDefault),
  "build-module": () => import('../chunks/build-module.mjs').then(_rDefault),
  cleanup: () => import('../chunks/cleanup.mjs').then(_rDefault),
  clean: () => import('../chunks/cleanup.mjs').then(_rDefault),
  preview: () => import('../chunks/preview.mjs').then(_rDefault),
  start: () => import('../chunks/preview.mjs').then(_rDefault),
  analyze: () => import('../chunks/analyze.mjs').then(_rDefault),
  generate: () => import('../chunks/generate.mjs').then(_rDefault),
  prepare: () => import('../chunks/prepare.mjs').then(_rDefault),
  typecheck: () => import('../chunks/typecheck.mjs').then(_rDefault),
  usage: () => import('../chunks/usage.mjs').then(_rDefault),
  info: () => import('../chunks/info.mjs').then(_rDefault),
  init: () => import('../chunks/init.mjs').then(function (n) { return n.i; }).then(_rDefault),
  create: () => import('../chunks/init.mjs').then(function (n) { return n.i; }).then(_rDefault),
  devtools: () => import('../chunks/devtools.mjs').then(_rDefault),
  upgrade: () => import('../chunks/upgrade.mjs').then(_rDefault),
  test: () => import('../chunks/test.mjs').then(_rDefault),
  add: () => import('../chunks/add.mjs').then(_rDefault),
  new: () => import('../chunks/add.mjs').then(_rDefault)
};
function defineNuxtCommand(command) {
  return command;
}

export { commands as c, defineNuxtCommand as d };
